### DISCUSSION TITLE
Is it possible to establish connection via socket?

### DISCUSSION THREAD

**Discussion by hasantayyar** [cyrilgdn/terraform-provider-postgresql#113] [Q&A]
I see the the provider converts all host and port inputs to `postgresq://...` connection string format. 

Is it possible to establish connection via sockets?

The go library requires the socket directory and the filename as hostname without `postgres://` schema.

**Comment by cyrilgdn** [comment]
Hi @hasantayyar ,

I don't think it's possible currently :thinking: 
Normally we should be able to set the (url espaced) socket path as the host (e.g.: `postgresql://%2Fvar%2Flib%2Fpostgresql/dbname`) as mentioned in Postgres documentation, but `lib/pq` does not support it apparently (cf https://github.com/lib/pq/issues/796).

But we could probably add the possibility to pass it as a parameter (e.g.: `postgres://username:@/dbname?host=/tmp/run`) with a specific `socket` scheme.

I'll create an issue for that.

**Reply by hasantayyar** [reply]
Thanks for the answer. For some reason Github API is failing while marking this as an answer of this topic so I couldn't mark yet.
